http/src/v0/root_files/tar_helper.rs

```diff
@@ -0,0 +1,345 @@
+use super::GetError;
+use bytes::{buf::BufMut, Bytes, BytesMut};
+use ipfs::unixfs::ll::Metadata;
+use std::borrow::Cow;
+use std::path::{Path, PathBuf};
+use tar::{EntryType, Header};
+
+/// Tar helper is internal to `get` implementation. It uses some private parts of the `tar-rs`
+/// crate to append the headers and the contents to a pair of `bytes::Bytes` operated in a
+/// round-robin fashion.
+pub(super) struct TarHelper {
+    bufsize: usize,
+    written: BytesMut,
+    other: BytesMut,
```

Comment:
this was my initial thought and it would make sense assuming that hyper would do polling of
 1. writing of response
 2. polling of response stream

`Bytes` has no API to validate this, even if it did it'd be quite tricky. I am thinking of just simplifying these to one buffer, remove all `std::mem::swap` and hope for the best.

Reply:
Done